lib/matplotlib/cbook/__init__.py

```diff
@@ -2075,3 +2081,14 @@ def _check_and_log_subprocess(command, logger, **kwargs):
             .format(command, exc.output.decode('utf-8')))
     logger.debug(report)
     return report
+
+
+def _check_enum(values, **kwargs):
```

Comment:
Not quite sure about the function name. This is not talking to me what it's doing. Maybe `_check_contains()` or `_check_in_list()`?

Reply:
It checks that it's a valid enumeration value, though I agree it's not a great name.

check_in_list looks backwards (btw, the reason for the `**kwargs` API is not to be able to squeeze multiple checks in a single call, but because that looks less bad (IMO) than `_check_enum("foo", foo, [...])`) and check_contains doesn't really convey the right idea to me (yes, strictly speaking, it's a containment check, but I think of it more as a validity check).

Anyways I don't feel very strongly about it but I'll wait until another reviewer chimes in so that we can agree on a name -- even though it's just a sed replace, I also need to realign the line continuations under them and I'm too lazy to do this twice.

Reply:
We have to bring in the name, the values and the allowed values somehow. I don't really have a good solution either. So just brainstorming:

It get's a bit better with kwargs, but also more verbose:
~~~
cbook._check_param('which', value=which, allowed_values=['tick1', 'tick2', 'grid'])
~~~

`rcsetup.py` does something similar with `ValidateInStrings`.

~~~
cbook._check_enum(['tick1', 'tick2', 'grid'], which=which)
~~~

would be

~~~
ValidateInStrings('which', ['tick1', 'tick2', 'grid'])(which)
~~~
Similarly, one could some factory function:
~~~
cbook._arg('which', ['tick1', 'tick2', 'grid']).validate(which)
# or
cbook._checker('which', ['tick1', 'tick2', 'grid']).check(which)
~~~
That's a bit more clear with respect to the actual action and it's target, at the cost of a slightly more complex implementation.

Alternatively one could define a `namedtuple('Param', 'name supported_vals')` and

~~~
cbook._check_param(which, Param('which', ['tick1', 'tick2', 'grid']))
~~~

Reply:
If we're really going that route, I'd rather keep the argument and its name together (`_checker([...])("foo", foo)`) but that's quite close to overengineering it IMO.

Reply:
Well, maybe `_check_enum("foo", foo, [...])` isn't too bad after all.